app/controllers/phylo_trees_controller.rb

```diff
@@ -127,10 +127,8 @@ def retry
 
   def download_snps
     snp_file = Tempfile.new
-    s3_file = @phylo_tree.s3_outputs["SNP_annotations"]
-    cmd_status = Open3.capture3("aws", "s3", "cp", s3_file, snp_file.path)[2]
-    unless cmd_status.success?
-      snp_file.write("Not yet available.")
+    s3_file = @phylo_tree.snp_annotations
+    unless s3_file && Open3.capture3("aws", "s3", "cp", s3_file, snp_file.path)[2].success?
```

Comment:
we have various download function from PipelineRun, reuse?

Reply:
ok. The download functions in PipelineRun are a bit different in the arguments (destination directory instead of file name) and behavior (create a new file, whereas here I want to use the tempfile I created previously). I will move my code to PipelineRunsHelper so it can be reused in the future.